test/sys/test_socket.rs

```diff
@@ -1462,3 +1462,40 @@ pub fn test_vsock() {
     close(s1).unwrap();
     thr.join().unwrap();
 }
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[test]
+pub fn test_sockaddr_storage_to_datalink() {
+    use nix::sys::socket::sockaddr_storage;
+    // in C, sockaddr_* are as if sockaddr_storage was a union
+    let mut storage : sockaddr_storage = unsafe { std::mem::zeroed::<sockaddr_storage>() };
+    let ll: *mut libc::sockaddr_ll = &mut storage as *mut _ as *mut libc::sockaddr_ll;
+    unsafe {
+        // NOTE: libc has libc::AF_PACKET as cint, instead of u16
+        (*ll).sll_family = libc::AF_PACKET as u16;
+        // NOTE: libc has libc::ETH_P_IP as cint, instead of u16
+        (*ll).sll_protocol = (libc::ETH_P_IP as u16).to_be();
+        (*ll).sll_ifindex = 1i32;
+        (*ll).sll_hatype = libc::ARPHRD_ETHER.to_be();
+        (*ll).sll_pkttype = 0u8; // PACKET_HOST (constant not in libc crate)
+        (*ll).sll_halen = 6;
+        // 00:00:5E:00:53:00 is reserved for documentation.
```

Comment:
This comment still doesn't match the code.

Reply:
Hmm, I'd argue that it's the 00: form that is technically reserved, but yes, you're right this is confusing.  Fixed.

Reply:
No, it's not fixed.  What did you even change?  I can't tell, because you force pushed.

Reply:
Urgh, my humblest apologies.

I'm still trying to learn rust, and the github workflows, please bear with me.   Last time you wanted everything squashed before commit, which I managed to completely screw up multiple times, so this time I've been trying to keep the commit simplified as I go along so hopefully I don't quite screw it up as badly...

I think this is correct this time.